From e156725317e37587a7dc9df9d9d4fe5748dd8ec9 Mon Sep 17 00:00:00 2001
From: Zhipeng Gong <zhipeng.gong@intel.com>
Date: Thu, 26 Jul 2018 09:46:38 +0800
Subject: [PATCH 2366/2367] drm/i915/gvt: notify global gtt update through g2v

This patch extends g2v notification to notify host GVT-g of
ggtt update from guest, including ggtt_insert_entries and
ggtt_clear_range. This patch also add one new pvmmio level
to control ggtt update.

This patch is needed for both uos and sos.

v2:
- calculate num_entries from gtt_entries.

v3:
- pass cache_level to GVT-g
- consolidate the same code into a function.

Tracked-On: projectacrn/acrn-hypervisor#994
Signed-off-by: Zhipeng Gong <zhipeng.gong@intel.com>
Reviewed-by: He, Min <min.he@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_gtt.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_gem_gtt.c b/drivers/gpu/drm/i915/i915_gem_gtt.c
index 98a8511..adbd61a 100644
--- a/drivers/gpu/drm/i915/i915_gem_gtt.c
+++ b/drivers/gpu/drm/i915/i915_gem_gtt.c
@@ -2152,6 +2152,11 @@ static void gen8_ggtt_insert_page(struct i915_address_space *vm,
 		return;
 	}
 
+	if (PVMMIO_LEVEL_ENABLE(vm->i915, PVMMIO_GGTT_UPDATE)) {
+		vgpu_ggtt_insert(vm->i915, offset, 1, level);
+		return;
+	}
+
 	ggtt->invalidate(vm->i915);
 }
 
-- 
2.7.4

